tests/test_E4_verification.py

```diff
@@ -35,3 +37,71 @@ def test_minimal_renewable_share_test_fails():
     }
     return_value = E4.minimal_renewable_share_test(dict_values)
     assert return_value == False
+
+
+def test_detect_excessive_excess_generation_in_bus_warning_is_logged(caplog):
+    """A logging.warning is printed due to excess generation. """
+    bus_label = "Test_bus"
+    dict_values = {
+        "optimizedFlows": {
+            bus_label: pd.DataFrame(
+                {"inflow": [1, 2, 3], "outflow": [-1, -1, -2], "excess": [0, 1, 1]}
+            )
+        }
+    }
+    with caplog.at_level(logging.WARNING):
+        E4.detect_excessive_excess_generation_in_bus(dict_values=dict_values)
+    assert (
+        f"Attention, on bus {bus_label} there is an excessive excess generation"
+        in caplog.text
+    ), f"An intended warning is not logged although there is excess generation."
+
+
+def test_detect_excessive_excess_generation_in_bus_no_excess(caplog):
+    """No excess generation takes place. """
```

Comment:
No excess generation? But there is excess. You mean "no excessive excess generation"?

Reply:
oh right, thanks a lot!